packages/amplify-provider-awscloudformation/src/utils/upload-auth-trigger-template.ts

```diff
@@ -0,0 +1,39 @@
+import { $TSContext, $TSObject, FeatureFlags, JSONUtilities, pathManager } from 'amplify-cli-core';
+import { S3 } from '../aws-utils/aws-s3';
+import path from 'path';
+
+export const AUTH_TRIGGER_TEMPLATE = 'auth-trigger-cloudformation-template.json';
+export const AUTH_TRIGGER_STACK = 'AuthTriggerCustomLambdaStack';
+
+export async function uploadAuthTriggerTemplate(context: $TSContext) {
+  if (!FeatureFlags.getBoolean('auth.breakCircularDependency')) {
+    return {};
+  }
+  const category = 'auth';
+  let { amplifyMeta } = context.amplify.getProjectDetails();
+  const authResource = amplifyMeta?.auth ?? {};
+  const authTriggerCfnFilePath = path.join(pathManager.getBackendDirPath(), category, Object.keys(authResource)[0], AUTH_TRIGGER_TEMPLATE);
+  try {
+    const cfnObject: $TSObject = JSONUtilities.readJson(authTriggerCfnFilePath);
+    const S3_UPLOAD_PATH = path.join(category, AUTH_TRIGGER_TEMPLATE);
+
+    return {
+      AuthTriggerTemplateURL: await uploadCfnToS3(context, S3_UPLOAD_PATH, cfnObject),
+    };
+  } catch {
+    return {
+      AuthTriggerTemplateURL: '',
+    };
+  }
+}
+
+async function uploadCfnToS3(context: any, cfnFile: string, cfnData: object): Promise<string> {
```

Comment:
I didnt find any related function

Reply:
we have this function https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-provider-awscloudformation/src/push-resources.ts#L684
might need to be refactored a bit to fit the use case here, but we shouldn't duplicate the upload functionality

Reply:
Looking a little close into the logic in this function, I think this template file is already uploaded as part of the push
https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-provider-awscloudformation/src/push-resources.ts#L635

Reply:
Yeah, I have  already added  trigger template glob ignore so, it will ignore the template when pushing others, but refactored the function to use the current function present in cli